citar-file.el

```diff
@@ -82,6 +82,22 @@ separator that does not otherwise occur in citation keys."
                  (const :tag "Find files with space after key" "[[:space:]]")
                  (regexp :tag "Filename separator")))
 
+(defcustom citar-file-open-external-extension-alist '(("html" . system))
```

Comment:
Note: we don't need a new defcustom; we need to refactor the current one.

See my minimal example elsewhere on this PR, and on #507 ;-)

Reply:
I think there are four:
- open with system default  which will be easy handled by current `citar-open-external`
- open with a customized application which need a new function for each command or make `citar-open-external` accept some args (and it will make the `citar-file-open-functions-alist` a little complicated, for example, `("pdf" . #'citar-open-external "sumatrapdf")`
- open with a funciton as `citar-file-open-functions-alist` does

Are you happy with somethink like `("pdf" . #'citar-open-external "sumatrapdf")`. If so, I just need to refactor my current code.

Reply:
Let's see what input @roshanshariff might have. He's been working on related code more recently than I, I believe, and I've been distracted with some IRL things.

I do wonder, however (it's a genuine question) if your `("pdf" . #'citar-open-external "sumatrapdf")` example could just be a lambda instead for the `cdr`?

Reply:
It could be a lambda, but I don't expect user might be familar with lambda or elisp funciton.

Reply:
Yes, that's true.

Reply:
@ShuguangSun - your example of `sumatrapdf` reminded me of a discussion on the org list awhile back, where someone was wanting to use `sioyek` (which I'd never heard of). This was his solution; just another real example to check to make sure we have this right:

```elisp
(setq citar-file-open-function '(lambda (file)
 (call-process "sioyek" nil 0 nil
 file)))
```